storage-node/packages/runtime-api/index.js

```diff
@@ -84,225 +91,247 @@ class RuntimeApi {
     return this.asyncLock.acquire(`${accountId}`, func)
   }
 
-  /*
-   * Wait for an event. Filters out any events that don't match the module and
-   * event name.
-   *
-   * The result of the Promise is an array containing first the full event
-   * name, and then the event fields as an object.
-   */
-  async waitForEvent(module, name) {
-    return this.waitForEvents([[module, name]])
-  }
-
-  static matchingEvents(subscribed, events) {
-    debug(`Number of events: ${events.length} subscribed to ${subscribed}`)
-
+  static matchingEvents(subscribed = [], events = []) {
     const filtered = events.filter((record) => {
-      const { event, phase } = record
-
-      // Show what we are busy with
-      debug(`\t${event.section}:${event.method}:: (phase=${phase.toString()})`)
-      debug(`\t\t${event.meta.documentation.toString()}`)
+      const { event } = record
 
       // Skip events we're not interested in.
       const matching = subscribed.filter((value) => {
-        return event.section === value[0] && event.method === value[1]
+        if (value[0] === '*' && value[1] === '*') {
+          return true
+        } else if (value[0] === '*') {
+          return event.method === value[1]
+        } else if (value[1] === '*') {
+          return event.section === value[0]
+        } else {
+          return event.section === value[0] && event.method === value[1]
+        }
       })
       return matching.length > 0
     })
-    debug(`Filtered: ${filtered.length}`)
 
-    const mapped = filtered.map((record) => {
+    return filtered.map((record) => {
       const { event } = record
       const types = event.typeDef
-
-      // Loop through each of the parameters, displaying the type and data
-      const payload = {}
+      const payload = new Map()
       event.data.forEach((data, index) => {
-        debug(`\t\t\t${types[index].type}: ${data.toString()}`)
-        payload[types[index].type] = data
+        const type = types[index].type
+        payload.set(index, { type, data })
       })
 
       const fullName = `${event.section}.${event.method}`
+      debug(`matched event: ${fullName} =>`, ...event.data.map((data) => `${data},`))
```

Comment:
Perhaps `event.data.join(', ')` would be a little cleaner

Reply:
indeed it would, will update.

Reply:
Fixed in https://github.com/Joystream/joystream/pull/1063/commits/ed574fc389cea6a653979320eb4ee30999463e23